packages/neuron-wallet/src/utils/to-csv-row.ts

```diff
@@ -41,7 +41,12 @@ const toCSVRow = (
       txType = +tx.sudtInfo.amount <= 0 ? `UDT ${SEND_TYPE}` : `UDT ${RECEIVE_TYPE}`
     }
   } else {
-    amount = shannonToCKB(BigInt(tx.value ?? ''))
+    amount = new Intl.NumberFormat('en-US', {
+      useGrouping: false,
+      signDisplay: 'always',
+      minimumFractionDigits: ckbDecimals,
+      maximumFractionDigits: ckbDecimals,
+    }).format(Number(formatUnit(BigInt(tx.value ?? '0'), 'ckb')))
```

Comment:
`Number` has precision issue, e.g.

```
new Intl.NumberFormat('en-US', {
      useGrouping: false,
      signDisplay: 'always',
      minimumFractionDigits: 8,
      maximumFractionDigits: 8,
    }).format(Number('123456789012345678901234567890'))
// +123456789012345680000000000000.00000000
```

`formatUnit` returns `string` which is enough for this case.

Reply:
It's better to enclose the `type cast` in a function and guard it by test cases.

I just checked the function of `foramtUnit`(https://github.com/ckb-js/lumos/blob/bb1fd0a233ac69ce1976e6b9036ea7fa17476457/packages/bi/src/index.ts#L197) and found that this method transformed `shannon` to `CKB` well. Why is `Intl.NumberFormat` used here without `usingGrouping`?

If `Intl.NumberFormat` is just for the positive sign, it can be fixed easily.

Reply:
`Intl.NumberFormat` is used for two reasons
1. Add sign `+` when necessary
2. Make sure `amount` has 8 fraction digits. 
e.g. `formatUnit(BigInt(0), 'ckb')` returns '0.0' instead of '0.00000000' 
and `formatUnit(BigInt(12345678900000000), 'ckb')` returns `"123456789.0"` instead of `"+123456789.00000000"`

The `useGrouping: false` option makes sure `amount` has no commas.

Here are the `shannonToCKB.test.ts` test cases
```ts
const fixtures: { shannon: bigint; expected: string }[] = [
  {
    shannon: BigInt(0),
    expected: '+0.00000000',     // Received: "0.0" without Intl.NumberFormat
  },
  {
    shannon: BigInt(1234567),
    expected: '+0.01234567',
  },
  {
    shannon: BigInt(-1234567),
    expected: '-0.01234567',
  },
  {
    shannon: BigInt(12345678),
    expected: '+0.12345678',
  },
  {
    shannon: BigInt(-12345678),
    expected: '-0.12345678',
  },
  {
    shannon: BigInt(123456789),
    expected: '+1.23456789',
  },
  {
    shannon: BigInt(-123456789),
    expected: '-1.23456789',
  },
  {
    shannon: BigInt(12345678900000000),
    expected: '+123456789.00000000',  // Received: "123456789.0" without Intl.NumberFormat
  },
  {
    shannon: BigInt(-12345678900000000),
    expected: '-123456789.00000000',  // Received: "-123456789.0" without Intl.NumberFormat
  },
]
```

Reply:
Got it, then I would suggest adding a method to enclose the `type cast`

Reply:
```ts
```

Reply:
Maybe it is good enough to rewrite the `NumberFormat` declaration for this scenario

```ts
declare global {
  namespace Intl {
    interface NumberFormat {
      format(value: number | string): string;
    }
  }
}
```

Reply:
Where to put this declaration ? When I try to put it inside `to-csv-row.ts`
<img width="440" alt="image" src="https://github.com/nervosnetwork/neuron/assets/7459812/5f5168f7-008a-4978-af0d-67f2db39c8eb">


I actually think `as any` or `as unknown as number` is not that bad because at langage level the `Intl.NumberFormat.prototype.format()` method does accept string. 

It's the typings are not well written, or too strict.

<img width="547" alt="image" src="https://github.com/nervosnetwork/neuron/assets/7459812/dddbd614-88a1-4143-87e5-81971dd55404">

Reply:
> Where to put this declaration ? When I try to put it inside to-csv-row.ts

![image](https://github.com/nervosnetwork/neuron/assets/164372891/5776568d-060d-4bb0-ab8f-bb47c0f94f7b)

Reply:
> Where to put this declaration ? When I try to put it inside to-csv-row.ts

![image](https://github.com/nervosnetwork/neuron/assets/7511174/0bb90535-d412-421a-81ed-2e06d10041ca)

Reply:
OK. Cool!  @Keith-CY What do you think about this appoach ?

Reply:
> All dirty works are concealed in the formatAmount method, and it can be guaranteed by unit tests.

Doing it in this way does indeed make it easier to read, without relying on TypeScript's tricks

Reply:
> > Where to put this declaration ? When I try to put it inside `to-csv-row.ts` <img alt="image" width="440" src="https://private-user-images.githubusercontent.com/7459812/320774482-5f5168f7-008a-4978-af0d-67f2db39c8eb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2NTE2ODAsIm5iZiI6MTcxMjY1MTM4MCwicGF0aCI6Ii83NDU5ODEyLzMyMDc3NDQ4Mi01ZjUxNjhmNy0wMDhhLTQ5NzgtYWYwZC02N2YyZGIzOWM4ZWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDlUMDgyOTQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Yzk2OWY2ZjEzZDliN2Q1NjFkMmNjN2IxYzVjMGE3NjZhMzZiMjY4Yjc0MDk2NzFiOTM2ODMyNjdhZmFkMzJmMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.C-ntLhpKS0Wi00oPYYx4mAqrQ-OyZgzjPDi9yMs--s8">
> > I actually think `as any` or `as unknown as number` is not that bad IN THIS CASE because at langage level the `Intl.NumberFormat.prototype.format()` method does accept string.
> > It's the typings are not well written, or too strict.
> > <img alt="image" width="547" src="https://private-user-images.githubusercontent.com/7459812/320774680-dddbd614-88a1-4143-87e5-81971dd55404.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2NTE2ODAsIm5iZiI6MTcxMjY1MTM4MCwicGF0aCI6Ii83NDU5ODEyLzMyMDc3NDY4MC1kZGRiZDYxNC04OGExLTQxNDMtODdlNS04MTk3MWRkNTU0MDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDlUMDgyOTQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDg2ODhlODE3ZGE1YzBjYTc2OTllMDQ2MmIxN2NhNzYzM2EyNmUwODg4ZDExNzVlNmMzYjI4ODlhZTk0MGY4OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.nqWI3Qvd0T2dr9Jc9myUT8BXh9yJoE64TqOlt9LYAHM">
> > @Keith-CY Can you show a little example about how to enclose the `type cast` by adding a method in this case ? I'm a bit confused.
> 
> Simply add a method as
> 
> ```ts
> const formatAmount = (amount: string | number | ...) => 
>     new Intl.NumberFormat('en-US', {
>       useGrouping: false,
>       signDisplay: 'always',
>       minimumFractionDigits: ckbDecimals,
>       maximumFractionDigits: ckbDecimals,
>     }).format(formatUnit(BigInt(amount ?? '0'), 'ckb') as any)
> ```
> 
> All dirty works are concealed in the `formatAmount` method, and it can be guaranteed by unit tests.

OK! I will do it.